### ISSUE TITLE
[Feature] performance alignment with vllm when server Qwen 7B model

### ISSUE THREAD

**Issue by yangwhale** [sgl-project/sglang-jax#183]
### Checklist

- [ ] 1. If the issue you raised is not a feature but a question, please raise a discussion at https://github.com/sgl-project/sgl-jax/discussions/new/choose Otherwise, it will be closed.
- [ ] 2. Please use English, otherwise it will be closed.

### Motivation

This is a real customer requirement and evaluating between vllm and sglang.

The goal of this task is to optimize the inference performance (measured in Requests Per Second, RPS) of the sgl_jax Qwen/Qwen2.5-7B-Instruct model running on v6e-1.

Objectives and Constraints

Target RPS: >10 (equivalent to 600 RPM), to achieve a price-performance ratio comparable to the Nvidia L20.
Inference Workload: 3000 input tokens, 30 output tokens (source).
Latency Constraints:
Time to First Token (TTFT) + 16 * Time Per Output Token (TPOT) < 500ms
End-to-end latency < 800ms

vllm benchmark as below:
python3 -m sgl_jax.bench_serving \
	--backend vllm \
	--dataset-name random \
	--num-prompts 500 \
	--random-input 3000 \
	--random-output 30 \
	--max-concurrency 8 \
	--random-range-ratio 1 \
	--warmup-requests 0

============ Serving Benchmark Result ============
Backend:                                 vllm      
Traffic request rate:                    inf       
Max request concurrency:                 8         
Successful requests:                     392       
Benchmark duration (s):                  39.05     
Total input tokens:                      1176000   
Total generated tokens:                  11760     
Total generated tokens (retokenized):    11755     
Request throughput (req/s):              10.04     
Input token throughput (tok/s):          30117.25  
Output token throughput (tok/s):         301.17    
Total token throughput (tok/s):          30418.43  
Concurrency:                             7.94      
----------------End-to-End Latency----------------
Mean E2E Latency (ms):                   791.40    
Median E2E Latency (ms):                 791.99    
---------------Time to First Token----------------
Mean TTFT (ms):                          188.80    
Median TTFT (ms):                        187.36    
P99 TTFT (ms):                           410.65    
---------------Inter-Token Latency----------------
Mean ITL (ms):                           20.79     
Median ITL (ms):                         20.65     
P95 ITL (ms):                            22.31     
P99 ITL (ms):                            22.71     
Max ITL (ms):                            65.15     
==================================================

### Related resources

TPU_CHIPS_PER_HOST_BOUNDS=1,1,1 VLLM_TPU_BUCKET_PADDING_GAP=128 VLLM_USE_V1=1 vllm serve Qwen/Qwen2.5-7B --seed 42 --disable-log-requests --gpu-memory-utilization 0.95 --max-num-batched-tokens 640 --max-num-seqs 256 --max-model-len 3072 --no-enable-prefix-caching
